### ISSUE TITLE
 Proton 5.13 Asian fonts not visible

### ISSUE THREAD

**Issue by hjc73wr** [ValveSoftware/steam-runtime#289]
#### Please describe your issue in as much detail as possible:

![2020-10-28_01-13-02](https://user-images.githubusercontent.com/18324708/97337220-f6e33080-18ba-11eb-88ed-a6d36c158860.png)
![2020-10-28_01-13-15](https://user-images.githubusercontent.com/18324708/97337231-fb0f4e00-18ba-11eb-9abd-53e50364e109.png)

I fixed temporarily the problem by create link
`ln -s /run/host/usr/share/fonts ~/.local/share/fonts`

![2020-10-28_01-14-01](https://user-images.githubusercontent.com/18324708/97337565-69eca700-18bb-11eb-87f5-753f353ef58e.png)
![2020-10-28_01-14-16](https://user-images.githubusercontent.com/18324708/97337576-6d802e00-18bb-11eb-82f7-87c359ffda9c.png)

**Comment by TTimo** [comment]
Hello @hjc73wr, please provide the output from 'Help' > 'System Information' - see https://github.com/ValveSoftware/steam-runtime/blob/master/doc/reporting-steamlinuxruntime-bugs.md for details.

**Comment by hjc73wr** [comment]
system information:
https://gist.github.com/hjc73wr/ea43e31018de15b9a355c2f3fc554d99

pressure-vessel.log
```Running Steam on ubuntu 20.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Steam client's requirements are satisfied
/home/user/.steam/debian-installation/ubuntu12_32/steam -nominidumps -nobreakpad
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
[2020-10-28 13:37:47] Startup - updater built Oct  7 2020 06:15:36
[2020-10-28 13:37:47] Loading cached metrics from disk (/home/user/.steam/debian-installation/package/steam_client_metrics.bin)
[2020-10-28 13:37:47] Using the following download hosts for Public, Realm steamglobal
[2020-10-28 13:37:47] 1. http://media4.steampowered.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2020-10-28 13:37:47] Verifying installation...
[2020-10-28 13:37:47] Verification complete
[2020-10-28 13:38:25] Shutdown```

**Comment by TTimo** [comment]
Thanks - you mentioned a workaround:

`ln -s /run/host/usr/share/fonts ~/.local/share/fonts`

Is this something you are doing at host level, or in the soldier container? This reads like a host operation - I don't have an ubuntu system to check against atm, but I don't think `/run/host/usr/share/fonts` normally exists on the host - it becomes a valid path in the soldier container maybe.

Possible explanation: Proton isn't checking `/run/host/usr/share/fonts` for fonts, but it does check `~/.local/share/fonts`.

**Comment by hjc73wr** [comment]
Yes, `/run/host/usr/share/fonts` doesn't exist on the host. I create link on host, but it is used for Soldier container.
![2020-10-29_16-50-16](https://user-images.githubusercontent.com/18324708/97548282-e392ab00-1a09-11eb-9abc-b25797238e43.png)
![2020-10-29_16-54-13](https://user-images.githubusercontent.com/18324708/97548294-e7bec880-1a09-11eb-80f8-41020c734d8c.png)

Fonts is work in `/usr/share/fonts`, `/usr/local/share/fonts`, `~/.local/share/fonts` and `~/.fonts`
![2020-10-29_17-09-43](https://user-images.githubusercontent.com/18324708/97548647-5a2fa880-1a0a-11eb-817c-f8a0e4d71763.png)

But I checked Proton script, it seems to only check built-in fonts.
```
def __init__(self, base_dir):
    self.base_dir = base_dir + "/"
    self.dist_dir = self.path("dist/")
    self.bin_dir = self.path("dist/bin/")
    self.lib_dir = self.path("dist/lib/")
    self.lib64_dir = self.path("dist/lib64/")
    self.fonts_dir = self.path("dist/share/fonts/")
    self.version_file = self.path("version")
    self.default_pfx_dir = self.path("dist/share/default_pfx/")
    self.user_settings_file = self.path("user_settings.py")
    self.wine_bin = self.bin_dir + "wine"
    self.wineserver_bin = self.bin_dir + "wineserver"
    self.dist_lock = FileLock(self.path("dist.lock"), timeout=-1)
```

**Comment by smcv** [comment]
The pressure-vessel container runtime system does not currently support using fonts that are not included in the runtime itself. Presumably we don't include enough fonts to have Asian character set coverage.

Using `~/.local/share/fonts` is OK as a temporary workaround, but we should do at least one of these two things:

* get fonts from the host system, and configure libraries inside the container to use them, like Flatpak does
* include fonts with wider character set coverage in the Runtime (`fonts-noto` maybe)

**Comment by smcv** [comment]
> get fonts from the host system, and configure libraries inside the container to use them, like Flatpak does

This is implemented and should be in the next pressure-vessel release.

**Comment by TTimo** [comment]
Should be fixed in soldier 0.20201124.0. See https://steamcommunity.com/app/221410/discussions/2/2962768718547168164/ for details.

**Comment by hjc73wr** [comment]
`soldier 0.20201124.0` solved this problem, thanks.
